### ISSUE TITLE
Make dialog component accessible

### ISSUE THREAD

**Issue by schanzer** [bootstrapworld/codemirror-blocks#347]
- [ ] aria-label for kbd shortcuts with punctuation should read the punctuation aloud (`Shift-,` shoudl be "shift comma", `esc`->"escape", `ctrl`->"control"). Need a hidden span with the speakable name, put the `<kbd>` element under aria-hidden
- [x] dialogs need to be fully modal (trap tab, shift-tab, heading steals focus)
- [x] dialog constructor should take a heading, tabIndex=-1, aria-labeledBy for dialog should be the heading
